plutus-chain-index/app/Marconi/Orphans.hs

```diff
@@ -0,0 +1,26 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
```

Comment:
I agree in principle, but then the original simple function that @andreabedini had starts to sound quite appealing.

Reply:
I agree in principle too. I think a monomorphic function `prettyChainPoint :: ChainPoint -> Doc ann` would be the best trade off because we are not really using the polymorphism anywhere (i.e. we are not passing a `ChainPoint` to anything that is `Pretty a => a`).
Also this is not a library but an executable, nobody can import this.

Reply:
We actually use the `Pretty` instance with `PrettyObject` to nicely integrate the `LogMsg` with `Pretty`, so `Pretty` actually simplifies things a bit.

@andreabedini This is actually a library, not an executable. I would expect a new package like `marconi-pab` which imports `marconi` and indexes stuff specific for the Contract API. Also, with the query interface, someone actually needs to import `marconi` if he wants to query the information that's stored in memory.

Reply:
following up in private

Reply:
I prefer the way it is written right now, to be honest.